### DISCUSSION TITLE
Speedport W504V

### DISCUSSION THREAD

**Discussion by BojanSof** [Freetz-NG/freetz-ng#986] [Allgemein]
I have Speedport W504V router, manufactured by AVM (not the Arcadyan one).
This is some info that I've gathered from inspecting the PCB and via telnet:

PCB inspection:
```
DSL-PHY: PSB80PGP (Infineon)
DSL-SoC: PSB7531ZDW (Infineon)
FPGA: XILINX Spartan XC3S100E
DECT SoC: SC14488-ANF80VE
Flash: 16 MB (MX29GL128EHT2I-90G)
RAM: 64 MB (2x MT48LC16M16A2-6A)
Ethernet: 4 ports (Switch IC: Atheros AR8216-BH1A)
WLAN: yes (IC: Atheros AR9220-AC1A)
USB: yes
```
Via telnet:
```
  kernel: Linux speedport.ip 2.6.32.21 #1 Tue Apr 3 15:38:38 CEST 2012 mips GNU/Linux
  firmware: 92 . 05.04 -22376
  Flash partitioning:
  mtd0    0x90000000,0x90000000
  mtd1    0x90020000,0x90F80000
  mtd2    0x90000000,0x90020000
  mtd3    0x90F80000,0x90FC0000
  mtd4    0x90FC0000,0x91000000
```

Also some part from `/var/env`:
```
HWRevision      164
HWSubRevision   2
ProductID       Fritz_Box_SpeedportW504V
SerialNumber    0000000000000000
annex   B
autoload        yes
bootloaderVersion       1.853
bootserport     tty0
cpufrequency    360000000
firstfreeaddress        0x946BF720
firmware_version        avme
firmware_info   92.05.04
flashsize       0x01000000
jffs2_size      50
memsize 0x04000000
modetty0        38400,n,8,1,hw
modetty1        38400,n,8,1,hw
mtd0    0x90000000,0x90000000
mtd1    0x90020000,0x90F80000
mtd2    0x90000000,0x90020000
mtd3    0x90F80000,0x90FC0000
mtd4    0x90FC0000,0x91000000
my_ipaddress    192.168.178.1
prompt  Eva_AVM
ptest
req_fullrate_freq       120000000
sysfrequency    120000000
urlader-version 1853
```

It has ADAM2 bootloader, which I can access via FTP first seconds after boot.

It looks very similar to Speedport W503V, which has half the Flash of W504V, and similar to FritzBox 7270v3 by comparing the components on the PCB. The difference is the FPGA used in 7270v3 compared to W504V.

I've tried building 7270v3 firmware image using Freetz-NG and flashing the image via FTP (bootloader):
```
binary
debug
quote MEDIA FLSH
put img_name.image mtd1
quote REBOOT
```

Sadly, shortly after boot, the DSL LED on the router was blinking twice, in regular intervals.
I've restored the original firmware from the dumps I've made and it came back functional.

I want to install Freetz on this router. Does anyone know if that is possible? I couldn't find much info on this router on the internet (there is more about Typ C - the one from Arcadyan).

Any help is much appreciated! Thank you in advance.

**Comment by fda77** [comment]
Afair for those old devices changes of kerne modules are needed, also things like https://github.com/Freetz-NG/freetz-ng/tree/master/patches/devices/04_XX/W501V
And you need a serial console to see the error messages, see also https://openwrt.org/toh/t-com/speedport_w_504v
Or simply install openwrt

**Reply by BojanSof** [reply]
The problem is that the one on openwrt is for the Arcadyan variant, not the one from AVM. It has different main SoC.

**Comment by fda77** [comment]
:(
You could also search www.ip-phone-forum.de for this device, there are some very old posts.
And ask "hippie" of "boxmatrix" in its irc channel. he has some non-public patches, maybe for 504v too

**Reply by BojanSof** [reply]
Thank you very much for the suggestions. I appreciate them.

**Comment by BojanSof** [comment]
Big thanks for the suggestions @fda77, with help from hippie, we created this PR: https://github.com/Freetz-NG/freetz-ng/pull/988.
